### DISCUSSION TITLE
Cannot push next.js project to github because large .next/cache files

### DISCUSSION THREAD

**Discussion by c2bsolutions** [vercel/next.js#41567] [Help]
Hello,

I am having issues pushing a next js project to github due to the below error. I have tried ignoring and deleting the files and using git-lfs to track the files to avoid the error, but all my attempts have been unsuccessful. I found this discussion with the same issue but it has gone unresolved -> https://github.com/vercel/next.js/discussions/29233

Does anyone have a resolution for this error?

remote: warning: File .next/cache/webpack/client-development/index.pack.old is 73.64 MB; this is larger than GitHub's recommended maximum file size of 50.00 MB
remote: warning: File .next/cache/webpack/client-development/12.pack is 61.65 MB; this is larger than GitHub's recommended maximum file size of 50.00 MB
remote: warning: File .next/cache/webpack/client-development/9.pack is 63.60 MB; this is larger than GitHub's recommended maximum file size of 50.00 MB
remote: warning: File .next/cache/webpack/client-production/0.pack is 88.83 MB; this is larger than GitHub's recommended maximum file size of 50.00 MB
remote: warning: File .next/cache/webpack/client-development/index.pack is 73.64 MB; this is larger than GitHub's recommended maximum file size of 50.00 MB
remote: warning: File .next/cache/webpack/client-development/21.pack is 62.56 MB; this is larger than GitHub's recommended maximum file size of 50.00 MB
remote: warning: File .next/cache/webpack/client-development/7.pack is 66.98 MB; this is larger than GitHub's recommended maximum file size of 50.00 MB
remote: warning: File .next/cache/webpack/client-development/index.pack is 73.64 MB; this is larger than GitHub's recommended maximum file size of 50.00 MB
remote: warning: File .next/cache/webpack/client-development/index.pack.old is 73.64 MB; this is larger than GitHub's recommended maximum file size of 50.00 MB
remote: warning: File .next/cache/webpack/client-production/9.pack is 55.30 MB; this is larger than GitHub's recommended maximum file size of 50.00 MB
remote: warning: File .next/cache/webpack/client-development/25.pack is 62.44 MB; this is larger than GitHub's recommended maximum file size of 50.00 MB
remote: error: Trace: d4d828f15f1df9a3c905902455c7dc946017c5d660254b4577da082475068581
remote: error: See http://git.io/iEPt8g for more information.
remote: error: File .next/cache/webpack/client-development/28.pack is 124.65 MB; this exceeds GitHub's file size limit of 100.00 MB
remote: error: File .next/cache/webpack/client-development/1.pack is 155.89 MB; this exceeds GitHub's file size limit of 100.00 MB
remote: error: File .next/cache/webpack/client-production/0.pack is 144.56 MB; this exceeds GitHub's file size limit of 100.00 MB
remote: error: File .next/cache/webpack/client-development/0.pack is 325.31 MB; this exceeds GitHub's file size limit of 100.00 MB
remote: error: GH001: Large files detected. You may want to try Git Large File Storage - https://git-lfs.github.com.

Thanks in Advance!

**Comment by gaston-flores** [comment]
Did you try to delete the files in your git history like @ijjk [mentioned here](https://github.com/vercel/next.js/discussions/29233#discussioncomment-1358739)? You may need to delete them using a tool like [BFG](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/removing-sensitive-data-from-a-repository#using-the-bfg) to completely remove them from git.

Also, did you add `.next` to gitignore?

**Reply by c2bsolutions** [reply]
Hello Gaston,

Yes, .next is in being ignored in the gitignore but when pushing to github the .next files are still staged. I removed the .cache files also but I think the files are coming from another branch or prior commit. Is there a best practice to clear all branches and reset to remove all files and start over?

Thanks in Advance,

**Reply by gaston-flores** [reply]
@c2bsolutions If you don't mind losing **everything** in your git history, deleting the `.git` directory in your project might be the easiest way to get rid of the error. 

A better alternative is using [BFG](https://rtyley.github.io/bfg-repo-cleaner/) to remove those files:

> The BFG is a simpler, faster alternative to [git-filter-branch](https://git-scm.com/docs/git-filter-branch) for cleansing bad data out of your Git repository history:
> - Removing **Crazy Big Files**
> - Removing Passwords, Credentials & other Private data

**Comment by thiswallz** [comment]
remove the folder from github ( you can do it from the UI, there is a delete button when you select the folder  ), then pull the changes to your local env,  modify gitignore, and push again. Should be fine now , second options is the same first part, but cloning again

**Comment by ohabash** [comment]
isnt .next/ required for starting prod app?
